src/common/safeIpc.ts

```diff
@@ -63,6 +63,7 @@ export interface SendChannels {
     'file-export-template': SendChannelInfo;
     'finish-loading': SendChannelInfo;
     'installing-deps': SendChannelInfo;
+    'updating-deps': SendChannelInfo;
```

Comment:
Instead of adding a new channel and duplicating the `.on(handler)` code, could you add a parameter to `installing-deps` instead?
```suggestion
    'installing-deps': SendChannelInfo<[onlyUpdating: boolean]>;
```